### ISSUE TITLE
note/comment component

### ISSUE THREAD

**Issue by jaketrent** [pluralsight/design-system#528]
Design (Web): https://www.figma.com/file/YsTklBecdddy9RZ3HXddIseD/PSDS-Web?node-id=2366%3A310 
Design (specced): https://www.figma.com/file/TbG4IGHd8th48OGQiQxYAwxb/DS-Note-Component

No nested comments

Todo:
- [x] Followup with devtective on their usage

**Comment by jaketrent** [comment]
Triage: need design review

**Comment by jaketrent** [comment]
For the design review:

```
## Note design review

- Called "note"?
- Avatar - just the one size?
- Clarify: Avatar and "Creator Name"/title always appear *together*?
- Message data type?
- Metadata:
  - data type?
  - limit on items?
- actionBar 
  - data type? 
  - Note.Action a la Card?
  - limit on actions?
- Edit state part of DS? Separate component?
  - Don't want to ship edit bits to read-only experiences
  - Perhaps separate EditNote component?

#### Proposed API

```
<Note 
  avatar={<Avatar name="Jake Trent" src="http://img.jpg" />}
  message={}
  metadata={[ , ]}
  actionBar={}
/>
```

- Avatar size overridden internally
- Avatar name drives the Note title
```

I plan on bringing this up in standup today.  I'm hoping this will speed up the conversation to fit in that mtg. cc @danethurber , @dborodyansky .

**Comment by jaketrent** [comment]
Design review outcomes:
- Ask devtective if they need more than strings.
- Message *maybe* as children
- Metadata as in Card
- Metadata no limit in count
- actionBar as in Card, locked viz, etc
- 2 Actions max
- No edit case
- Explicit heading prop
- Semantics on heading={<h3>string only/name</h3>}, no link
- Build this for threaded convo that devtective is working on; mult teams in content tools plan to use
- Defer onClick on the whole region; no anchor/href support;

**Comment by jaketrent** [comment]
- devtective - node, because multi-paragraphs, formatting over lines, workflow comments
- message={node} - Text.P x 1-n example
- margin from parent; first-child remove top margin; last-child remove bottom margin; rely on collapse to not double up spacing
- linkable - ask devtective, otherwise .x release when someone asks for it
- meta flex space handled like Card/Row, nowrap
- Divider in stack of notes follow css rules of Row, border is on top of each item (except for first)
- List semantics - 
<Note.List>
<Note />
<Note />
<Note />
</Note.List>
- TODO: Dmitry contact devtective

**Comment by dborodyansky** [comment]
@getschomp Per our slack chat on reviewing the standard note/comment component design and API, would you mind giving a thumbs up or any commentary right here in the issue? Thanks!

**Comment by getschomp** [comment]
```
Message data type? NODE, display, likely HTML, client to use dangerouslySetInnerHtml on child el
```

Our PM wanted us to use markdown eventually, our audience should be well versed in markdown.  We went with plain text in the interim since we were not sure how the design system component would be implemented.  If we do use html for note structuring we will need to also render html code samples as well as the html for note structuring.  

Additionally, we need text to go to a new line where we store a new line character in our database.

```
Edit state part of DS? Separate component? NO, no edit to be built in DS
Don't want to ship edit bits to read-only experiences
Perhaps separate EditNote component?
```
We went with two components `Note` and `NoteEditor`.
https://github.com/ps-dev/assessment-tools/blob/master/scribe/client/components/CommentPanel/CommentPanel.jsx

```
Dmitry: does dev-tective want linkable avatar and heading?
```
This wasn't part of our requirements

```
limit on actions? YES, 2
```
This works for us

FWIW our current component looks like this:

![Screen Shot 2019-05-22 at 4 53 44 PM](https://user-images.githubusercontent.com/5835824/58208283-6ca0a100-7cb2-11e9-9979-fb5833d80d12.png)
